pkg/bundle/internal/target/target.go

```diff
@@ -242,6 +256,23 @@ func (r *Reconciler) needsUpdate(ctx context.Context, kind Kind, log logr.Logger
 		needsUpdate = true
 	}
 
+	// Get old password hash annotations.
+	// Calculate new password hash annotation with changed password.
+	// Compare old password hash annotations with new ones.
+	if bundle.Spec.Target.AdditionalFormats != nil {
+		if bundle.Spec.Target.AdditionalFormats.JKS != nil {
+			if hash, ok := obj.GetAnnotations()[JksPasswdHashAnnotation]; !ok || hash != fmt.Sprintf("%x", sha256.Sum256([]byte(*bundle.Spec.Target.AdditionalFormats.JKS.Password))) {
+				needsUpdate = true
+			}
+		}
+
+		if bundle.Spec.Target.AdditionalFormats.PKCS12 != nil {
+			if hash, ok := obj.GetAnnotations()[Pkcs12PasswdHashAnnotation]; !ok || hash != fmt.Sprintf("%x", sha256.Sum256([]byte(*bundle.Spec.Target.AdditionalFormats.PKCS12.Password))) {
+				needsUpdate = true
+			}
+		}
+	}
+
```

Comment:
This block seems overly complicated. Why can't you use the same (simple) approach as for the data hash:

https://github.com/cert-manager/trust-manager/blob/17561571e008ac2775df6ef966bf89dd325a9913/pkg/bundle/internal/target/target.go#L241

Reply:
Indeed, it could be simple, however in some conditions when `Password` is nil application is falling into `nil pointer dereference` panic. _Though modesty be a virtue, yet bashfulness is a vice._  :)

Reply:
But why do you need to calculate any hash here? The correct approach is just to check if the annotations are present with the correct value. And the correct value is the hash calculate in the reconcile func - based on the desired state.

Reply:
Unfortunately, then no chance to identify that was a password  changed, so no bundle update is done.

Reply:
Ah, I guess you are talking about putting hashes values to `SyncConfigMap` function.

Reply:
This should be very similar to what is currently done to detect bundle content changes (data hash). I am trying to argue that what you are working on should be aligned with the existing functionality. 😉 I would probably supply the desired annotations (containing at least the existing data hash annotation, and eventually these new truststore password annotations) to the sync-functions. The code should be fairly easy to check if the target resource already has the correct annotations - with the correct values (hashes).

Reply:
Fixed.

Reply:
````go
for k, v := range passwdHashAnnotations {
    if obj.GetAnnotations()[k] != v {
      needsUpdate = true
    }
}
````

Isn't something like this simpler?

Reply:
Well, for many items a loop should be used with not doubt, however we have only 3 items now, so for obviousness I prefer don't use it.
Anyway, to simplify the code I did as you proposed.